### PR TITLE
PHOENIX-7040 (ADDENDUM): Support TTL for views using the new column TTL in SYSTEM.CATALOG

### PR DESCRIPTION
**Pull request by lokiore**
- Adding full more tests for ttl at different levels in heirarchy
- Also resolving bug related to using physical link for getting ttl for views while building PTable.

### CODE REVIEW

**Comment by lokiore** [comment]
FYI @jpisaac

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java [hunk 1]
```diff
@@ -1617,66 +1620,98 @@ private PTable getTableFromCells(List<Cell> tableCellList, List<List<Cell>> allC
         return builder.build();
     }
 
+    /**
+     * Method to return TTL value defined at current level ot up the Hierarchy of the view.
```
Comment (jpisaac) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java:1624]: nit: typo "or"?

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java [hunk 2]
```diff
@@ -1617,66 +1620,98 @@ private PTable getTableFromCells(List<Cell> tableCellList, List<List<Cell>> allC
...
+     * @param viewKey Key of the view for which we have to find TTL
+     * @param clientTimeStamp Client TimeStamp
+     * @return TTL value for a given view, if nothing is defined anywhere then return
+     *         TTL_NOT_DEFINED(0).
+     * @throws IOException
+     * @throws SQLException
+     */
+
```
Comment (jpisaac) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java:1632]: nit: scanTTLFromParent should be called getTTLFromHierarchy WDT?

**Review by jpisaac — COMMENTED** [review]
LGTM, Couple of nits

**Review by jpisaac — APPROVED** [review]

**Pull request merged**